### ISSUE TITLE
nav2 devcontainer on humble branch uses rolling image

### ISSUE THREAD

**Issue by Ryanf55** [ros-navigation/navigation2#3790]
## Bug report

**Required Info:**

- Operating System:
  -  Ubuntu 23
- ROS2 Version:
  - Docker (Ros humble)
- Version or commit hash:
  - humble
- DDS implementation:
  - n/a
#### Steps to reproduce issue

``` 
git checkout humble
# Open vscode devcontainer and observe the compile failure from trying to compile humble in a rolling environment
```

#### Expected behavior

The devcontainer uses a `humble` base image for Docker.

#### Actual behavior

The argument for the ROS version is unset, and defaults to `rolling` regardless of which branch of nav2 you use.

In the dockerfile:
`ARG FROM_IMAGE=ros:rolling`

I would expect `FROM_IMAGE` is set to `humble` when using VSCode on the humble branch.

#### Additional information

Relates to https://github.com/microsoft/vscode-remote-release/issues/3545.
The recommended path from that thread is to use docker compose to supply those options.

**Comment by SteveMacenski** [comment]
@ruffsl

**Comment by Ryanf55** [comment]
Please assign this to me; I can take the recommended route to add compose support to nav2. Do you want a `docker-compose.yml` file at the root of the repo, or in the `.devcontainer` folder?

**Comment by ruffsl** [comment]
> I would expect `FROM_IMAGE` is set to `humble` when using VSCode on the humble branch.

How would you expect to set the build arg dynamically, while remaining agnostic of the docker client? The dev container could be started from Mac or windows, so use of scripts would be out of scope. We could just change/commit the default value for that build arg per branch.

**Comment by Ryanf55** [comment]
> > I would expect `FROM_IMAGE` is set to `humble` when using VSCode on the humble branch.
> 
> How would you expect to set the build arg dynamically, while remaining agnostic of the docker client? The dev container could be started from Mac or windows, so use of scripts would be out of scope. We could just change/commit the default value for that build arg per branch.

That's what I did; take a look at the draft PR?

**Comment by Ryanf55** [comment]
I tried a very specific change to just change the `build.cacheFrom` argument, however the dockerfile still has an ARG for the ROS distro, and VSCode doesn't support build arguments in devcontainer. 
https://github.com/microsoft/vscode-remote-release/issues/3545

I don't have a good path forward on this.

**Comment by Ryanf55** [comment]
We have a path forward. I'd say once the next VSCode version is out of beta and released, we can add build arguments to specify a ros distro.
https://github.com/microsoft/vscode-remote-release/issues/3545#issuecomment-1892274755

**Comment by tonynajjar** [comment]
This should now be possible @Ryanf55 right?

**Comment by Ryanf55** [comment]
> This should now be possible @Ryanf55 right?

Correct, feel free to submit a PR. I have other priorities right now.

**Comment by SteveMacenski** [comment]
Working on this right now as a Friday afternoon toy project. I just enabled Jazzy in https://github.com/ros-navigation/navigation2/pull/5119 and https://github.com/ros-navigation/navigation2/pull/5121. Working on Humble next https://github.com/ros-navigation/navigation2/pull/5122
